### PR TITLE
Fixing a minor memory issue in the discrete ordinates solver destructor

### PR DESCRIPTION
**Pull request by wdhawkins**
Debugging this issue led to some minor cleanup in lbs_groupset. One problem being that not all of the lbs_groupset member variables were properly initialized. It also led to a small fix in runtime::Finalize().

### CODE REVIEW

**Review by ragusa — APPROVED** [review]

File: modules/linear_boltzmann_solvers/lbs_solver/groupset/lbs_groupset.h [hunk 1]
```diff
@@ -68,23 +68,30 @@ class LBSGroupset : public Object
...
+
   /**Constructs the groupset subsets.*/
   void BuildSubsets();
 
-public:
   /**Constructs the groupset subsets.*/
   void PrintSweepInfoFile(size_t ev_tag, const std::string& file_name);
+
+private:
+  void init(int id);
```
Comment (andrsd) [inline modules/linear_boltzmann_solvers/lbs_solver/groupset/lbs_groupset.h:94]: `init` -> `Init`

**Review by andrsd — CHANGES_REQUESTED** [review]

File: modules/linear_boltzmann_solvers/lbs_solver/groupset/lbs_groupset.h (hunk 1 above)
Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/lbs_solver/groupset/lbs_groupset.h]: Fixed. Thanks!

**Review by andrsd — APPROVED** [review]

File: modules/linear_boltzmann_solvers/lbs_solver/groupset/lbs_groupset.cc [hunk 2]
```diff
@@ -15,19 +15,20 @@ namespace lbs
...
 {
   InputParameters params = Object::GetInputParameters();
 
   params.SetGeneralDescription("Input Parameters for groupsets.");
   params.SetDocGroup("LuaLBSGroupsets");
 
   params.AddRequiredParameterArray("groups_from_to",
-                                   "The first and last group id this groupset operates on."
+                                   "The first and last group id this groupset operates on "
                                    " e.g. A 4 group problem <TT>groups_from_to= {0, 3}</TT>");
```
Comment (zhardy-lanl) [inline modules/linear_boltzmann_solvers/lbs_solver/groupset/lbs_groupset.cc:27]:
```suggestion
                                   "The first and last group id this groupset operates on, "
                                   " e.g. <TT>groups_from_to= {0, 3}</TT> for a 4 group problem.");
```

**Review by zhardy-lanl — CHANGES_REQUESTED** [review]
Please remove the blank lines from the `GetInputParameters` routine for groupsets except to break logically related groups.

File: modules/linear_boltzmann_solvers/lbs_solver/groupset/lbs_groupset.cc (hunk 2 above)
Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/lbs_solver/groupset/lbs_groupset.cc]: Fixed.

**Review by zhardy-lanl — APPROVED** [review]

**Pull request merged**